ec/ecresource/deploymentresource/read.go

```diff
@@ -67,3 +80,15 @@ func readResource(_ context.Context, d *schema.ResourceData, meta interface{}) d
 
 	return diags
 }
+
+func deploymentNotFound(err error) bool {
+	// We're using the As() call since we do not care about the error value
+	// but do care about the error type since it's an implicit 404.
+	var notDeploymentNotFound *deployments.GetDeploymentNotFound
+	if errors.As(err, &notDeploymentNotFound) {
+		return true
+	}
+
+	// We also check for the case where a 403 is thrown for ESS.
```

Comment:
I realise ESS returns a 403 when a deployment doesn't exist, but I'm a bit concerned about the implications of doing this in the case that the user is getting a 403 for an actual authorisation issue 🤔